rest/data/patch.go

```diff
@@ -113,6 +115,27 @@ func (pc *DBPatchConnector) SetPatchActivated(patchId string, user string, activ
 	if err != nil {
 		return err
 	}
+	if activated && p.Version == "" {
+		ctx := context.Background()
```

Comment:
pass this in or TODO

you can also get a context from the environment (as well as the settings object, so it might be better to pass the environment rather than settings.)